mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -648,4 +648,50 @@ mod tests {
 
         assert_eq!(expected_user_agent, user_agent_header_value);
     }
+
+    #[test]
+    #[ignore]
+    // Simple test to ensure the user agent header is correct even when given profile contains calid credentials.
+    fn test_client_with_valid_credentials() {
+        let expected_user_agent = "mountpoint-s3-client";
+        let profile_name_override = String::from("VALID_CREDENTIALS");
+
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+
+        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");
+
+        let mut message = client
+            .new_request_template("GET", "plutotestankit")
```

Comment:
for bucket name, let's use something generic. maybe `some-bucket` or review another example.

put this in a constant in this test file. (probably better if we define this somewhere common TBH, but we can do that later.